sandy/fy.py

```diff
@@ -203,6 +210,92 @@ def _expand_zam(self):
                            dtype=int)
         return self.data.assign(Z=zam.Z, A=zam.A, M=zam.M)
 
+    def custom_perturbation(self, zam, mt, e, zap, pert):
+        """
+        Apply a custom perturbation in the fission yields identified by the mt,
+        for a given  zam in a given energy and for a given product.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which perturbations are to be
+            applied.
+        mt : `int`
+            MT reaction number of the FY to which perturbations are to be
+            applied.
+        e : 'float'
+            Energy to which perturbations are to be applied.
+        zap : `int`
+            Product identifier to which perturbations are to be
+            applied.
+        pert : 'float'
+            Perturation coeffcients as ratio values.
+
+        Returns
+        -------
+        `Fy`
+            Fission yield instance with given value ZAM/ZAP/MT/E perturbed.
+
+        Examples
+        --------
+        >>> tape = sandy.get_endf6_file("jeff_33",'nfpy','all')
+        >>> nfpy = Fy.from_endf6(tape)
+        >>> nfpy_pert = nfpy.custom_perturbation(922350, 459, 0.0253, 551370, -0.1)
+        >>> comp = nfpy_pert.data.query('ZAM==922350 & ZAP==551370 & MT==459 & E==0.0253').squeeze().FY
+        >>> assert np.setdiff1d(nfpy_pert.data.values,nfpy.data.values) == comp
+        """
+        df = self.data.copy()
+        mask = (df.ZAM == zam) & (df.ZAP == zap) & (df.MT == mt) & (df.E == e)
+        df.loc[mask, "FY"] = df[mask].squeeze().FY * (1 + pert)
+        return self.__class__(df)
+
+    def apply_bmatrix(self, zam, e, decay_data):
+        """
+        Apply calculated IFY with (1-B) method in a given zam
+        for a given energy.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which perturbations are to be
+            applied.
+        e : 'float'
+            Energy to which perturbations are to be applied.
+        decay_data : `sandy.DecayData`
+            Radioactive nuclide data for several isotopes.
+
+        Returns
+        -------
+        `Fy`
+            Fission yield instance with IFY calculated for a given value
+            of ZAM/E.
+
+        Examples
+        --------
+        >>> zam = [380900,551370,541350]
+        >>> decay_minimal = sandy.get_endf6_file("jeff_33", 'decay', zam)
```

Comment:
you cannot test with a B-matrix for only 3 nuclides

Reply:
I have changed the whole test, so that the IFYs of the 3 cores change to make it a valid test.